cms/private_media/signal_handlers.py

```diff
@@ -0,0 +1,88 @@
+from typing import TYPE_CHECKING, Any
+
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ImproperlyConfigured
+from django.db.models import IntegerField
+from django.db.models.functions import Cast
+from wagtail.models import Page, ReferenceIndex
+from wagtail.signals import page_published, page_unpublished, published, unpublished
+
+from cms.private_media.models import PrivateImageMixin
+from cms.private_media.utils import get_private_media_models
+
+if TYPE_CHECKING:
+    from django.db.models import Model
+
+
+def publish_media_on_publish(instance: "Model", **kwargs: Any) -> None:
+    """Signal handler to be connected to the 'page_published' and 'published'
+    signals for all publishable models. It is responsible for identifying any
+    privacy-controlled media used by the object, and ensuring that it is also
+    made public.
+    """
+    for model_class in get_private_media_models():
+        model_ct = ContentType.objects.get_for_model(model_class)
+        referenced_pks = (
+            ReferenceIndex.get_references_for_object(instance)
+            .filter(to_content_type=model_ct)
+            .annotate(int_object_id=Cast("to_object_id", output_field=IntegerField()))
+            .values_list("int_object_id", flat=True)
+            .distinct()
+        )
+        queryset = model_class.objects.filter(pk__in=referenced_pks)
```

Comment:
question: should this limit to instances where `_privacy=Privacy.PRIVATE`?
i.e. if referencing an already public image, do we need to process it again? Or better said, only bulk make public any items that are private, or public but outdated?

Reply:
This is something already checked by `bulk_make_private` and `bulk_make_public` before they make and updates, but it wasteful to fetch object that we know we don't need, so will apply!

Reply:
Addressed in [f7c46e0](https://github.com/ONSdigital/dis-wagtail/pull/46/commits/f7c46e0ac723804ca31151133198afaf9757b467)